desktop-environment.el

```diff
@@ -361,6 +389,27 @@ the screen."
     (async-shell-command desktop-environment-screenlock-command)))
 
 
+
+;;; Commands - keyboard layout
+
+;;;###autoload
+(defun desktop-environment-keyboard-layout-cycle-forward ()
+  "Cycle in desktop-environment-keyboard-layouts-ring by one forward."
+  (interactive)
+  ;; Cycle the layout ring by adding one to index. (modulo is done by ring-ref)
+  (setq desktop-environment-current-keyboard-layout-index (1+ desktop-environment-current-keyboard-layout-index))
+  (let ((async-shell-command-buffer 'new-buffer) (current-layout (ring-ref desktop-environment-keyboard-layouts-ring desktop-environment-current-keyboard-layout-index)))
+    (async-shell-command (concat "setxkbmap -layout " current-layout))))
```

Comment:
There is quite some duplicated code between the 2 commands. Can you please refactor? E.g.,

```emacs
(defun desktop-environment-keyboard-layout-cycle-forward ()
  "Cycle in desktop-environment-keyboard-layouts-ring by one forward."
  (interactive)
  (desktop-environment--keyboard-layout-change (1+ desktop-environment--current-keyboard-layout-index)))

(defun desktop-environment--keyboard-layout-change (new-index)
  "..."
  (setq desktop-environment-current-keyboard-layout-index new-index)
  ...)
```

Reply:
This seems much better, I didn't thought about it this way.